File: issuetracker/src/main/java/hu/elte/issuetracker/model/Issue.java
package hu.elte.issuetracker.model;

import com.sun.istack.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    @NotNull
    private String title;

    @Column
    @NotNull
    private String description;

    @Column
    @NotNull
    private String place;

    @Column
    @NotNull
    @Enumerated(EnumType.STRING)
    private Status status;

    @Column
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column
    @UpdateTimestamp
    private LocalDateTime modifiedAt;

    public enum Status {
        NEW, DOING, DONE
    }

    @ManyToOne
    private User creator;

    @ManyToMany
    private List<Label> labels;

    @OneToMany(mappedBy = "issue", cascade = CascadeType.REMOVE)
    private List<Message> messages;
}

File: issuetracker-client/src/app/domain/issue.ts
import { Label } from './label';
import { Message } from './message';
import { User } from './user';

export interface Issue {
  id: number;
  title: string;
  description: string;
  place: string;
  status: IssueStatus;
  createdAt: string;
  modifiedAt: string;
  labels: Label[];
  user: User;
  messages: Message[];
}

export enum IssueStatus {
  NEW = 'NEW',
  DOING = 'DOING',
  DONE = 'DONE',
}
